docs/singleKeyvault/README.md

```diff
@@ -0,0 +1,367 @@
+
+```
+$ az group create -l westus2 -n my-global-rg
+```
+
+### Create Storage Account in Azure
+
+Before attempting to deploy the infrastructure environments, you will also need to create an Azure Storage Account. You can do this in Azure Portal, or by using the Azure CLI:
+
+```
+az storage account create \
+    --name mystorageaccount \
+    --resource-group my-global-rg \
+    --location eastus \
+    --sku Standard_LRS \
+    --encryption blob
+```
+
+The Azure CLI needs your storage account credentials for most of the commands in this tutorial. While there are several options for doing so, one of the easiest ways to provide them is to set `AZURE_STORAGE_ACCOUNT` and `AZURE_STORAGE_KEY` environment variables.
+
+First, display your storage account keys by using the az storage account keys list command:
+
+```
+az storage account keys list \
+    --account-name myStorageAccount \
+    --resource-group my-global-rg \
+    --output table
+```
+
+Now, set the `AZURE_STORAGE_ACCOUNT` and `AZURE_STORAGE_KEY` environment variables. You can do this in the Bash shell by using the export command:
+
+```
+export AZURE_STORAGE_ACCOUNT="mystorageaccountname"
+export AZURE_STORAGE_KEY="myStorageAccountKey"
+```
+
+Blobs are always uploaded into a container. You can organize groups of blobs similar to the way you organize your files on your computer in folders.
+
+Create a container for storing blobs with the az storage container create command.
+
+```
+az storage container create --name mystoragecontainer
+```
+
+Next, let's create a folder called azure-common-infra
+```
+mkdir azure-common-infra 
+cd azure-common-infra
+```
+Next, use Bedrock cli command to scaffold the configuration for common-infra template using the following command. Here we are using the Bedrock’s predefined `azure-common-infra` template to create configuration parameters for westus cluster. 
+
+```
+spk infra scaffold --name westus --source https://github.com/microsoft/bedrock --version master --template cluster/environments/azure-common-infra
+```
+
+This `scaffold` command creates a directory called `westus` and creates a definition.yaml file in it that looks like this:
+
+```yaml
+name: westus
+source: 'https://github.com/microsoft/bedrock'
+template: cluster/environments/azure-common-infra
+version: master
+backend:
+  storage_account_name: <storage account name>
+  access_key: <storage access key>
+  container_name: <storage account container>
+  key: tfstate-common-infra
+variables:
+  address_space: <insert value>
+  keyvault_name: <insert value>
+  global_resource_group_name: <insert value>
+  service_principal_id: <insert value>
+  subnet_name: <insert value>
+  subnet_prefix: <insert value>
+  vnet_name: <insert value>
+```
+`scaffold` has downloaded the template locally, extracted all of the variables for the template, and provided defaults where possible for all of the variables.
+
+Let's fill in the variables for common-infra infrastructure variables. 
+
```

Comment:
Note: `global_resource_group_name` is the resource group that was created in [Resource Group Requirement](#resource-group-requirement).